pydis_site/apps/api/models/bot/user.py

```diff
@@ -86,3 +92,34 @@ def username(self) -> str:
         For usability in read-only fields such as Django Admin.
         """
         return str(self)
+
+class UserAltRelationship(ModelReprMixin, models.Model):
```

Comment:
I missed something from the original requirements when first reviewing this PR, only realised when starting to do the bot side.

We want an info/reason text field here too, which is required when marking two users as alts. This is so that mods can give proof as to _why_ they this the two users are alts.

This does also add a bit more complexity...as we likely want a way to edit this reason too.

I'll start on the bot side impl of this anyway, since it's mostly going to be the same from the bot's side

Reply:
I've implemented this, unfortunately we lose the symmetrical aspect of the `ManyToManyField` then:

https://docs.djangoproject.com/en/5.0/topics/db/models/#intermediary-manytomany

> For a model which has a many-to-many relationship to itself through an intermediary model, two foreign keys to the same model are permitted, **but they will be treated as the two (different) sides of the many-to-many relationship**. If there are more than two foreign keys though, you must also specify through_fields as above, or a validation error will be raised.

I've changed this such that our code (manually) adds records in both directions in a transaction, which should keep our database consistent. The only downside is that we are sort-of duplicating data. I am open to suggestions on other solutions for this.

Deletes are symmetrical automatically, I added tests to verify this works as expected for both additions and deletions in both ways.

Reply:
Oh, also note that the field names are not what you asked exactly because I've reused our existing model timestamp mixin. So you want to look at `created_at` and `updated_at` instead.

Reply:
Yea, that correct, clients should only be update to update the context, doing so will also update the updated_at field.